### DISCUSSION TITLE
GCNConv error on hetero graph

### DISCUSSION THREAD

**Discussion by 1AngelUS** [pyg-team/pytorch_geometric#5652] [Q&A]
Hi, 

I am implementing the following code from this https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html link.
I just replaced the GATConv layer with GCNConv layer. 
```
class HeteroGNN(torch.nn.Module):
    def __init__(self, hidden_channels, out_channels, num_layers):
        super().__init__()

        self.convs = torch.nn.ModuleList()
        for _ in range(num_layers):
            conv = HeteroConv({
                ('paper', 'cites', 'paper'): GCNConv(-1, hidden_channels),
                ('paper', 'rev_writes', 'author'): GATConv((-1, -1), hidden_channels),
            }, aggr='sum')
            self.convs.append(conv)

        self.lin = Linear(hidden_channels, out_channels)

    def forward(self, x_dict, edge_index_dict):
        for conv in self.convs:
            x_dict = conv(x_dict, edge_index_dict)
            x_dict = {key: x.relu() for key, x in x_dict.items()}
        return self.lin(x_dict['author'])

model = HeteroGNN(hidden_channels=64, out_channels=dataset.num_classes,
                  num_layers=2)
```

It is however throwing the following error:
**AttributeError: 'tuple' object has no attribute 'size'** in 1st GCNConv layer. Could you please help me why it is not working. 

Thank you!

**Comment by EdisonLeeeee** [comment]
Currently `GCNConv` is not supported for heterogeneous graphs. See [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/cheatsheet.html) for the supported layers on heterogeneous graphs.

**Reply by 1AngelUS** [reply]
Thank you for your quick reply!
I didn't know this. Thanks! 

They have used GCNConv in this link also, so, I got confused. 
https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html

**Reply by EdisonLeeeee** [reply]
In that example, GCNConv is only used when the source node and destination node share the same type. Actually, the relation `src == dst` is treated as a homogeneous one in HeteroConv:
https://github.com/pyg-team/pytorch_geometric/blob/a6b92fd8e3fcdaf2041e203343a966a264586069/torch_geometric/nn/conv/hetero_conv.py#L127-L131

**Reply by 1AngelUS** [reply]
Sorry for the late reply. 
Thank you for explaining it so well. I understand now :)
I will change GCNConv with some other GNN.
